permissions/views.py

```diff
@@ -0,0 +1,204 @@
+        team.courses.add(*courses)
+
+        team.save()
+
+    return redirect("permissions:home")
+
+
+@login_required()
+def team_detail(request, team_id):
+    users = User.objects.all()
+
+    context = {
+        "users": users
+    }
+
+    try:
+        team = Team.objects.get(id=team_id)
+        roles = Role.objects.filter(team=team)
+        permissions = Permission.objects.filter(team=team)
+        context["team"] = team
+        context["roles"] = roles
+        context["permissions"] = format_perm(permissions)
+        context["courses"] = team.courses.all()
+
+        role_map = {}
+
+        for member in team.members.all():
+            member_roles = list(
+                map(lambda x: x.name, member.role_set.filter(team=team)))
+            role_map[member.username] = ",".join(member_roles)
+
+        context["role_map"] = role_map
+
+    except Team.DoesNotExist:
+        return redirect("permissions:home")
+
+    return render(request, "team_page.html", context)
+
+
+@require_POST
+@login_required
+def create_role(request):
+    team_id = request.POST.get("team_id")
+    role_name = request.POST.get("role_name")
+    members_list = request.POST.getlist("members")
+
+    try:
+        team = Team.objects.get(id=team_id)
+        role = Role(
+            name=role_name,
+            created_by=request.user,
+            team=team
+        )
+
+        role.save()
+
+        members = User.objects.filter(username__in=members_list)
+        role.members.add(*members)
+
+        role.save()
+    except Team.DoesNotExist:
+        return redirect('permissions:home')
+
+    return redirect('permissions:team_detail', team_id)
+
+
+@require_POST
+@login_required
+def add_permission(request):
+    team_id = request.POST.get("team_id")
+    course_id = request.POST.get("courses")
+    role_id = request.POST.get("role")
+    perm_type = request.POST.get("perm_type")
+    units = request.POST.getlist("units")
+
+    team = Team.objects.get(id=team_id)
+    course = Course.objects.get(id=course_id)
+    role = Role.objects.get(id=role_id)
+
+    # Create permission obj for each unit
+    for unit in units:
+        type, id = unit.split("_")
+
+        if type == "quiz":
+            content_object = Quiz.objects.get(pk=id)
+        else:
+            content_object = Lesson.objects.get(pk=id)
+
+        permission = Permission(
+            team=team,
+            course=course,
+            perm_type=perm_type,
+            content_object=content_object
+        )
+
+        permission.save()
+
+        permission.role.add(role)
+
+        permission.save()
+
+    return redirect('permissions:team_detail', team_id)
+
+
+@login_required
+def delete_permission(request, permission_id, team_id):
+    '''
+    Delete permission.
+    Only team creator can delete
+    '''
+
+    try:
+        team = Team.objects.get(pk=team_id)
+
+        if team.created_by == request.user:
+            Permission.objects.get(pk=permission_id).delete()
```

Comment:
Will this work? I guess delete works on queryset and not on object.

Reply:
Yeah, this works. [reference](https://stackoverflow.com/questions/3805958/how-to-delete-a-record-in-django-models)